wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WMenuItemRenderer.java

```diff
@@ -1,20 +1,57 @@
 package com.github.bordertech.wcomponents.render.webxml;
 
+import com.github.bordertech.wcomponents.MenuSelectContainer;
 import com.github.bordertech.wcomponents.WComponent;
 import com.github.bordertech.wcomponents.WMenuItem;
+import com.github.bordertech.wcomponents.WebUtilities;
 import com.github.bordertech.wcomponents.XmlStringBuilder;
 import com.github.bordertech.wcomponents.servlet.WebXmlRenderContext;
-import com.github.bordertech.wcomponents.util.Util;
 
 /**
  * The Renderer for {@link WMenuItem}.
  *
  * @author Yiannis Paschalidis
+ * @author Mark Reeves
  * @since 1.0.0
  */
 final class WMenuItemRenderer extends AbstractWebXmlRenderer {
 
 	/**
+	 * A WMenuItem is selectable if it is explicitly set as selectable (or not selectable) or if its own selectability is not set if its nearest
+	 * neighbour selection container is selectable.
+	 * @param item the WMenuItem to test
+	 * @return {@code true} if the WMenuItem is selectable
+	 */
+	private boolean isSelectable(final WMenuItem item) {

```

Comment:
I can see the point of moving the `isSelected` logic to WMenuItem - even though it is a conversion from earlier client-only code. I do not believe we should do the same with `getRole` as this is **only** client code and is, in the first instance, an indication of the multi-selectability of the ancestor MenuContainer. We could output a selection mode attribute but I chose to output the role attribute as it will become useful when we move away from XSLT.